fails to build for ghc-9.6 stackage nightly

```
       Building library for calligraphy-0.1.4..                                                                                     
       [ 1 of 16] Compiling Calligraphy.Compat.GHC                                                                                  
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/calligraphy-0.1.4-64f21f0a1103f85c7459da488d68ab826301a1e9f8163a03b69831e07abadb2b/src
/Calligraphy/Compat/GHC.hs:57:1: error:                                                                                             
           Could not find module ‘GHC.Unit.Module.Name’                                                                             
           Perhaps you meant                                                                                                        
             GHC.Unit.Module.Env (from ghc-9.6.2)                                                                                   
             GHC.Unit.Module.Env (needs flag -package-id ghc-lib-parser-9.6.2.20230523)                                             
             GHC.Unit.Module.Deps (from ghc-9.6.2)                                                                                  
           Use -v (or `:set -v` in ghci) to see a list of the files searched for.                                                   
          |                                                                                                                         
       57 | import GHC.Unit.Module.Name                                                                                             
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                             
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/calligraphy-0.1.4-64f21f0a1103f85c7459da488d68ab826301a1e9f8163a03b69831e07abadb2b/src
/Calligraphy/Util/Printer.hs:16:37: error: [GHC-76037]                                                                              
           Not in scope: type constructor or class ‘Ap’                                                                             
          |                                                                                                                         
       16 |   deriving (Semigroup, Monoid) via (Ap Printer a)                                                                       
          |                                     ^^
```

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -57,7 +57,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: [ '8.8.4', '8.10.7', '9.0.2', '9.2.7', '9.4.4', '9.6.1' ]
+        ghc: [ '8.8.4', '8.10.7', '9.0.2', '9.2.7', '9.4.5', '9.6.2' ]
         os: [ ubuntu-latest, macos-latest ]
     runs-on: ${{ matrix.os }}
     steps:

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -18,6 +18,7 @@ jobs:
   nix-matrix:
     name: Nix ${{ matrix.ghc }}
     strategy:
+      fail-fast: false
       matrix:
         ghc: [ ghc8107, ghc902, ghc927, ghc945, ghc962 ]
     runs-on: ubuntu-latest

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -19,10 +19,7 @@ jobs:
     name: Nix ${{ matrix.ghc }}
     strategy:
       matrix:
-        # TODO 2023-04-14
-        # GHC 9.6 itself has hit nixpkgs, but many dependencies are still on older versions.
-        # As soon as the 9.6 nix build succeeds it should be added to the matrix.
-        ghc: [ ghc8107, ghc902, ghc927, ghc944 ]
+        ghc: [ ghc8107, ghc902, ghc927, ghc945, ghc962 ]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -1,8 +1,10 @@
 name: CI
 
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - 'master'
+  pull_request:
 
 jobs:
 

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -27,12 +27,12 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: cachix/install-nix-action@v19
-      - name: build shell
-        run: nix build -L .#devShells.x86_64-linux.${{ matrix.ghc }}-shell
-      - name: cabal build in shell
-        run: nix develop -L .#devShells.x86_64-linux.${{ matrix.ghc }}-shell -c cabal new-build
       - name: nix build
-        run: nix build -L .#apps.x86_64-linux.calligraphy-${{ matrix.ghc }}
+        run: nix build -L .#apps.x86_64-linux.calligraphyFor.${{ matrix.ghc }}
+      - name: build shell
+        run: nix build -L .#devShells.x86_64-linux.build-shells.${{ matrix.ghc }}
+      - name: cabal build in shell
+        run: nix develop -L .#devShells.x86_64-linux.build-shells.${{ matrix.ghc }} -c cabal new-build
 
   stack-build:
     name: Stack ${{ matrix.resolver }} on ${{ matrix.os }}

--- a/flake.nix
+++ b/flake.nix
@@ -24,7 +24,7 @@
         "ghc8107"
         "ghc884"
       ];
-      default-ghc-version = "ghc944";
+      default-ghc-version = "ghc945";
       per-compiler = fkey: fattr:
         (builtins.listToAttrs (builtins.map (str: { name = fkey str; value = fattr str; }) supported-ghc-versions))
         // { default = fattr default-ghc-version; };
@@ -45,7 +45,7 @@
                 hspkgs.cabal-install
                 hspkgs.haskell-language-server
                 hspkgs.hlint
-                hspkgs.ormolu
+                # hspkgs.ormolu
                 pkgs.bashInteractive
                 pkgs.graphviz
               ];

--- a/flake.nix
+++ b/flake.nix
@@ -6,6 +6,7 @@
 
   outputs = inputs:
     let
+
       overlay = final: prev: {
         haskell = prev.haskell // {
           packageOverrides = hfinal: hprev:
@@ -16,46 +17,54 @@
         calligraphy = final.haskell.lib.compose.justStaticExecutables final.haskellPackages.calligraphy;
       };
 
-      supported-ghc-versions = [
-        "ghc962"
-        "ghc945"
-        "ghc927"
-        "ghc902"
-        "ghc8107"
-        "ghc884"
-      ];
-      default-ghc-version = "ghc945";
-      per-compiler = fkey: fattr:
-        (builtins.listToAttrs (builtins.map (str: { name = fkey str; value = fattr str; }) supported-ghc-versions))
-        // { default = fattr default-ghc-version; };
 
       perSystem = system:
         let
+
+          per-compiler = f: pkgs.lib.genAttrs [
+            "ghc962"
+            "ghc945"
+            "ghc927"
+            "ghc902"
+            "ghc8107"
+            "ghc884"
+          ]
+            (ghc: f pkgs.haskell.packages.${ghc});
+
           pkgs = import inputs.nixpkgs { inherit system; overlays = [ overlay ]; };
-          mkApp = compiler:
-            pkgs.haskell.lib.compose.justStaticExecutables
-              pkgs.haskell.packages."${compiler}".calligraphy;
-          mkShell = compiler:
-            let hspkgs = pkgs.haskell.packages.${compiler}; in
+
+          mkApp = hspkgs: pkgs.haskell.lib.compose.justStaticExecutables hspkgs.calligraphy;
+
+          mkBuildShell = hspkgs:
+            hspkgs.shellFor {
+              packages = hpkgs: [ hpkgs.calligraphy ];
+              nativeBuildInputs = [ hspkgs.cabal-install ];
+            };
+
+          mkDevShell = hspkgs:
             hspkgs.shellFor {
               withHoogle = true;
               packages = hpkgs: [ hpkgs.calligraphy ];
-              GHC_VERSION = compiler;
               nativeBuildInputs = [
                 hspkgs.cabal-install
-                hspkgs.haskell-language-server
-                hspkgs.hlint
-                # hspkgs.ormolu
+                hspkgs.ormolu
                 pkgs.bashInteractive
                 pkgs.graphviz
+                hspkgs.hlint
+                hspkgs.haskell-language-server
               ];
             };
+
         in
         {
-          devShells = per-compiler (str: str + "-shell") mkShell;
-          packages.calligraphy = pkgs.calligraphy;
-          defaultPackage = pkgs.calligraphy;
-          apps = per-compiler (str: "calligraphy-" + str) mkApp;
+          packages.default = pkgs.calligraphy;
+
+          apps.default = pkgs.calligraphy;
+          apps.calligraphyFor = per-compiler mkApp;
+
+          devShells.default = mkDevShell pkgs.haskellPackages;
+          devShells.build-shells = per-compiler mkBuildShell;
+          devShells.dev-shells = per-compiler mkDevShell;
         };
     in
     { inherit overlay; } // inputs.flake-utils.lib.eachDefaultSystem perSystem;

--- a/flake.nix
+++ b/flake.nix
@@ -1,7 +1,7 @@
 {
   description = "calligraphy";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-23.05";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
   outputs = inputs:
@@ -45,9 +45,7 @@
                 hspkgs.cabal-install
                 hspkgs.haskell-language-server
                 hspkgs.hlint
-                # Ormolu doesn't work well with the way nix does old GHC
-                # versions, so I'm turning it off by default
-                # hspkgs.ormolu
+                hspkgs.ormolu
                 pkgs.bashInteractive
                 pkgs.graphviz
               ];

--- a/flake.nix
+++ b/flake.nix
@@ -17,8 +17,8 @@
       };
 
       supported-ghc-versions = [
-        "ghc961"
-        "ghc944"
+        "ghc962"
+        "ghc945"
         "ghc927"
         "ghc902"
         "ghc8107"

